src/System.Diagnostics.Process/tests/System.Diagnostics.Process.Tests/ProcessTests.cs

```diff
@@ -544,6 +544,33 @@ public void TestGetProcessesByName()
             Assert.True(Process.GetProcessesByName(currentProcess.ProcessName, currentProcess.MachineName).Count() > 0, "TestGetProcessesByName001 failed");
         }
 
+        public static IEnumerable<object[]> GetTestProcess()
+        {
+            Process currentProcess = Process.GetCurrentProcess();
+            yield return new object[] { currentProcess, Process.GetProcessById(currentProcess.Id, "127.0.0.1") };
```

Comment:
NotSupportedException is thrown only when accessing MainModule property on remote process. On a process on the same machine, it is either null or a ProcessModule.